modules/rdg/src/userobjects/ElementLoopUserObject.C

```diff
@@ -185,10 +185,31 @@ ElementLoopUserObject::onInternalSide(const Elem * elem, unsigned int side)
 }
 
 void
-ElementLoopUserObject::onInterface(const Elem * /*elem*/,
-                                   unsigned int /*side*/,
-                                   BoundaryID /*bnd_id*/)
+ElementLoopUserObject::onInterface(const Elem * elem, unsigned int side, BoundaryID /*bnd_id*/)
 {
+  _current_elem = elem;
+  // Pointer to the neighbor we are currently working on.
+  _current_neighbor = elem->neighbor(side);
+
+  // Get the global id of the element and the neighbor
+  const dof_id_type elem_id = elem->id();
+  const dof_id_type neighbor_id = _current_neighbor->id();
+
+  // TODO: add if-statement to check if this needs to be executed
+  if ((_current_neighbor->active() && (_current_neighbor->level() == elem->level()) &&
+       (elem_id < neighbor_id)) ||
+      (_current_neighbor->level() < elem->level()))
```

Comment:
I don't understand the logic here: In the first half of the expression (separated by the OR expression) you are checking (among other things for a lower ID), is that so that you only execute on the interface once? If so, what about the the second case, in this scenario the current_neighbor may not be active since you are just using the `neighbor` method of the current element which returns an element at the same level. I'm fairly certain this won't handle all adaptivity cases. Also, `neighbor` is deprecated.